packages/sprotty/src/features/routing/bezier-edge-router.ts

```diff
@@ -317,79 +317,64 @@ export class BezierEdgeRouter extends AbstractEdgeRouter {
  */
 export class BezierMouseListener extends MouseListener {
 
-    @inject(EdgeRouterRegistry) @optional() edgeRouterRegistry?: EdgeRouterRegistry;
-
     mouseDown(target: SModelElement, event: MouseEvent): Action[] {
-        if (!(target instanceof SRoutingHandle) || (target.kind !== 'bezier-add' && target.kind !== 'bezier-remove')) {
-            return [];
-        }
-
         const result = [];
-        let router;
-        const element = target.root.index.getById(target.id);
-        if (this.edgeRouterRegistry && element instanceof SRoutingHandle) {
-            router = this.edgeRouterRegistry.get((element.parent as SRoutableElement).routerKind);
-        }
-        if (router instanceof BezierEdgeRouter) {
+        if (target instanceof SRoutingHandle && (target.kind === 'bezier-add' || target.kind === 'bezier-remove')) {
             if (target.type === 'bezier-create-routing-point') {
-                result.push(new BezierCurveAction(router, BezierActionTask.ADD, target.pointIndex, target.parent.id));
+                result.push(new AddRemoveBezierSegmentAction(AddRemoveBezierSegmentActionTask.ADD, target.id));
             } else if (target.type === 'bezier-remove-routing-point') {
-                result.push(new BezierCurveAction(router, BezierActionTask.REMOVE, target.pointIndex, target.parent.id));
+                result.push(new AddRemoveBezierSegmentAction(AddRemoveBezierSegmentActionTask.REMOVE, target.id));
             }
         }
         return result;
     }
 };
 
-export enum BezierActionTask {
+export enum AddRemoveBezierSegmentActionTask {
```

Comment:
I'd use a simple string with constrained type: `'add' | 'remove'`

Reply:
Yes, constraining a type like this makes enums "obsolete" in this case. 👍